ndb/src/google/cloud/ndb/model.py

```diff
@@ -3676,6 +3676,80 @@ class a different name when stored in Google Cloud Datastore than the
         """
         return cls.__name__
 
+    def __hash__(self):
+        """Not implemented hash function.
+
+        Raises:
+            TypeError: Always, to emphasize that entities are mutable.
+        """
+        raise TypeError("Model is not immutable")
```

Comment:
Double negative.

Reply:
Sort of, but don't you think it's the "right" concept. I.e. `__hash__` can only be defined if "Model is immutable"